### PR TITLE
added python precommits to .travis.yml

### PR DESCRIPTION
**Pull request by TeodoroFilippini**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,41 +1,52 @@
...
+        # Merge all coverage reports located in subdirectories and put them under: gover.coverprofile
+        - gover
 
     # Send code coverage report to coveralls.io
-- goveralls -service=travis-ci -coverprofile=gover.coverprofile
\ No newline at end of file
+    - goveralls -service=travis-ci -coverprofile=gover.coverprofile
+    - language: python
+      python:
+        - "3.5"
```
Comment (mristin) [inline .travis.yml]: You can pick only a single python version (3.5), since this is not a python library.

File: .travis.yml [hunk 2]
```diff
@@ -1,41 +1,52 @@
...
+        - "3.6"
+      install:
+        - pip3 install -e .[dev]
+        - pip3 install coveralls
+      script:
+        - ./tests/precommit.py
+        - coveralls
```
Comment (mristin) [inline .travis.yml]: Can you try to run the component test as well?

File: .travis.yml [hunk 3]
```diff
@@ -1,41 +1,52 @@
...
-    # Merge all coverage reports located in subdirectories and put them under: gover.coverprofile
-    - gover
+      after_success:
+        # Merge all coverage reports located in subdirectories and put them under: gover.coverprofile
+        - gover
 
     # Send code coverage report to coveralls.io
-- goveralls -service=travis-ci -coverprofile=gover.coverprofile
\ No newline at end of file
+    - goveralls -service=travis-ci -coverprofile=gover.coverprofile
```
Comment (mristin) [inline .travis.yml]: I think the indention is wrong here?

File: .travis.yml [hunk 4]
```diff
@@ -1,41 +1,52 @@
...
+    - language: python
+      python:
+        - "3.5"
+        - "3.6"
+      install:
+        - pip3 install -e .[dev]
+        - pip3 install coveralls
```
Comment (mristin) [inline .travis.yml]: We don't need coveralls for python since this is not a python library (and hence no test coverage).

File: .travis.yml (hunk 2 above)
Comment (mristin) [inline .travis.yml]: We don't need coveralls for python since this is not a python library (and hence no test coverage).

File: .travis.yml [hunk 5]
```diff
@@ -1,41 +1,52 @@
...
     # Send code coverage report to coveralls.io
-- goveralls -service=travis-ci -coverprofile=gover.coverprofile
\ No newline at end of file
+    - goveralls -service=travis-ci -coverprofile=gover.coverprofile
+    - language: python
+      python:
+        - "3.5"
+        - "3.6"
+      install:
+        - pip3 install -e .[dev]
```
Comment (mristin) [inline .travis.yml]: I think you need to call:

```
pip3 install -r ./tests/requirements.txt
```
since there is no setup.py in the root.

**Comment by mristin** [comment]
You can have a look at the results at: https://travis-ci.com/Parquery/mailgun-relayery

**Comment by mristin** [comment]
The coveralls.io should be on as well.

File: .travis.yml (hunk 2 above)
Comment (TeodoroFilippini) [reply .travis.yml]: It needs a released version of the binaries to exist. I could write a script releasing to a temp dir and running the component test there, but would it be platform-independent?

File: .travis.yml (hunk 5 above)
Comment (TeodoroFilippini) [reply .travis.yml]: whoops, good catch, ty!

File: .travis.yml (hunk 2 above)
Comment (mristin) [reply .travis.yml]: It wouldn't, but travis runs on linux. I'd say it's worth a try, but if it doesn't work out-of-the-box, then let's just not bother.

**Pull request merged**